File: algorithm_java/src/main/java/programmers/lv2/모음사전.java
package programmers.lv2;

import java.util.HashMap;

public class 모음사전 {
    int answer = 0;
    int count = 0;
    String[] alpha = {"A", "E", "I", "O", "U"};

    public int solution(String word) {
        dfs("", word);

        return answer;
    }

    public void dfs(String nowWord, String target) {
        if (nowWord.equals(target)) {
            answer = count;
            return;
        }

        if (nowWord.length() == 5) {
            return;
        }

        for(int i = 0; i < 5; i++) {
            count++;
            dfs(nowWord + alpha[i], target);
        }
    }

}

File: algorithm/programmers/LV2/python/모음사전.py
order = 0


def dfs(now_word, dict, alpha):
    global order
    if len(now_word) > 5:
        return

    dict[now_word] = order
    order += 1

    for char in alpha:
        dfs(now_word + char, dict, alpha)


def solution(word):
    global order
    alpha = ['A', 'E', 'I', 'O', 'U']
    dict = {}

    dfs("", dict, alpha)

    return dict[word]


print(solution("AAA"))
